src/main/java/com/opentable/jvm/Nmt.java

```diff
@@ -0,0 +1,257 @@
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.opentable.jvm;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wrapper for Native Memory Tracking information.
+ * @see Memory#getNmt()
+ */
+public class Nmt {
+    private static final Logger LOG = LoggerFactory.getLogger(Nmt.class);
+    private static final String NMT_DISABLED = "Native memory tracking is not enabled\n";
+    // Data comes back in "KB", as formatted by the HotSpot VM.
+    // However, a read of the source code (specifically share/vm/utilities/globalDefinitions.hpp)
+    // reveals that they actually mean KiB.
+    @VisibleForTesting
+    static final long K = 1024;
+
+    public final Usage total;
+    // Keys are human-readable category names, such as "Java Heap" or "Arena Chunk".
+    // Categories' usages are not guaranteed to sum to total usage.
+    // Entries will be in the same order as the diagnostic command output.
+    public final Map<String, Usage> categories;
+
+    private Nmt(final Usage total, final Map<String, Usage> categories) {
+        this.total = total;
+        this.categories = categories;
+    }
+
+    /**
+     * Requires JVM argument -XX:NativeMemoryTracking=summary.
+     * Logs a warning if there was an error getting the NMT summary or if NMT was disabled.
+     * Available here in case you want easy access to the raw output from the VM (instead of our nice parse).
+     * @return Human-readable NMT summary.  null if there was an error getting the summary.
+     */
+    @Nullable
+    public static String invoke() {
+        final String ret = Dcmd.invoke("vmNativeMemory", "summary");
+        if (NMT_DISABLED.equals(ret)) {
+            LOG.warn(ret.trim());
+            return null;
+        }
+        return ret;
+    }
+
+    @Override
+    public String toString() {
+        return new Formatter().toString();
+    }
+
+    /**
+     * Requires JVM argument -XX:NativeMemoryTracking=summary.
+     * Logs a warning if there was an error getting the NMT summary or if NMT was disabled.
+     * @return null if there was an error getting the summary.
+     */
+    @Nullable
+    static Nmt get() {
```

Comment:
Maybe it's getting convoluted, but the idea is that `Nmt.invoke` won't be often-used.  Usually, you'll use `Memory.getNmt` or `Memory.formatNmt`.  Perhaps this isn't sufficiently clear, and some re-arranging is in order?